go-controller/pkg/ovn/base_network_controller_namespace.go

```diff
@@ -429,16 +435,24 @@ func (bnc *BaseNetworkController) createNamespaceAddrSetAllPods(ns string, ips [
 	return bnc.addressSetFactory.NewAddressSet(dbIDs, ips)
 }
 
+// createNamespacePortGroup should only create a port group if doesn't exist already,
+// all ports and acls will be added by pod/multicast/egressfirewall/etc handlers.
 func (bnc *BaseNetworkController) createNamespacePortGroup(ns string) (string, error) {
-	// Add all ports from this namespace to the multicast allow group.
-	ports, err := bnc.getNamespaceLSPs(ns)
-	if err != nil {
-		return "", fmt.Errorf("failed to create namespace port group")
-	}
-
+	var err error
 	portGroupName := bnc.getNamespacePortGroupName(ns)
-
-	pg := bnc.buildPortGroup(portGroupName, ns, ports, nil)
+	pg := &nbdb.PortGroup{
+		Name: portGroupName,
+	}
+	pg, err = libovsdbops.GetPortGroup(bnc.nbClient, pg)
+	if err != nil && !errors.Is(err, libovsdbclient.ErrNotFound) {
+		return "", fmt.Errorf("failed to get port group %s: %w", portGroupName, err)
+	}
+	if pg != nil {
+		// pg exists
+		return portGroupName, nil
+	}
+	// pg doesn't exist, create
+	pg = bnc.buildPortGroup(portGroupName, ns, nil, nil)
```

Comment:
We could have a `CreatePortGroup` method, just like `CreateAddressSets` or `CreateMeterBandOps` that just ensures the port group exists without updating it.

Reply:
good point, I completely forgot I can add new functions :confused: